modules/nf-core/svdb/merge/main.nf

```diff
@@ -3,57 +3,102 @@ process SVDB_MERGE {
-        'biocontainers/mulled-v2-c8daa8f9d69d3c5a1a4ff08283a166c18edb0000:511069f65a53621c5503e5cfee319aa3c735abfa-0' }"
+        'https://depot.galaxyproject.org/singularity/mulled-v2-375a758a4ca8c128fb9d38047a68a9f4322d2acd:b3615e06ef17566f2988a215ce9e10808c1d08bf-0':
+        'biocontainers/mulled-v2-375a758a4ca8c128fb9d38047a68a9f4322d2acd:b3615e06ef17566f2988a215ce9e10808c1d08bf-0' }"
 
     input:
     tuple val(meta), path(vcfs)
-    val (priority)
+    val(input_priority)
+    val(sort_inputs)
 
     output:
-    tuple val(meta), path("*.vcf.gz"), emit: vcf
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}"), emit: vcf
+    tuple val(meta), path("*.tbi")                    , emit: tbi, optional: true
+    tuple val(meta), path("*.csi")                    , emit: csi, optional: true
+    path "versions.yml"                               , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args   = task.ext.args ?: ''
+    def args2   = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def input  = "${vcfs.join(" ")}"
-    def prio   = ""
-    if(priority) {
-        prio = "--priority ${priority.join(',')}"
+
+    // Ensure priority list matches the number of VCFs if priority is provided
+    if (input_priority && vcfs.collect().size() != input_priority.collect().size()) {
+        error "If priority is used, one tag per VCF is needed"
+    }
+
+    def input = ""
+    def prio = ""
+    if (input_priority) {
+        if (vcfs.collect().size() > 1 && sort_inputs) {
+            // make vcf-prioprity pairs and sort on VCF name, so priority is also sorted the same
+            def pairs = vcfs.indices.collect { [vcfs[it], input_priority[it]] }
+            pairs = pairs.sort { a, b -> a[0].name <=> b[0].name }
+            vcfs = pairs.collect { it[0] }
+            priority = pairs.collect { it[1] }
+        } else {
+            priority = input_priority
+        }
+
+        // Build inputs
+        prio = "--priority ${input_priority.join(',')}"
         input = ""
-        for (int index = 0; index < vcfs.size(); index++) {
-            input += " ${vcfs[index]}:${priority[index]}"
+        for (int index = 0; index < vcfs.collect().size(); index++) {
+            input += "${vcfs[index]}:${priority[index]} "
         }
+
+    } else {
+        // if there's no priority input just sort the vcfs by name if possible
+        input = (vcfs.collect().size() > 1 && sort_inputs) ? vcfs.sort { it.name } : vcfs
     }
+
+    def extension = args2.contains("--output-type b") || args2.contains("-Ob") ? "bcf.gz" :
+                    args2.contains("--output-type u") || args2.contains("-Ou") ? "bcf" :
+                    args2.contains("--output-type z") || args2.contains("-Oz") ? "vcf.gz" :
+                    args2.contains("--output-type v") || args2.contains("-Ov") ? "vcf" :
+                    "vcf"
```

Comment:
Sorry, my update of this changed the output from vcf.gz to vcf.

Reply:
good catch

Reply:
I was planning to change all the bcftools / samtools modules and set the default for everyone to `vcf.gz` as required by nf-core guidelines.
Something to do for the Hackathon ? This would be a really good exercise for first timer.
There is many things to be done on this two group of modules 😉

Reply:
sure. I meant more because we are not being explicit in the outputs here the change could have very easily remained undetected :)

Reply:
No worries.
I think we mentioned somewhere that output should be in a compressed format by default, but as long as it's modifiable I'm happy enough for the day

Reply:
https://github.com/nf-core/modules/pull/6858 :)